### DISCUSSION TITLE
[match] is it possible to specify a keychain path instead of just keychain name?

### DISCUSSION THREAD

**Discussion by OliverKoo** [fastlane/fastlane#20556] [Q&A]
When running match is it possible to specify the path of the keychain where you want the certs to be install to?
I only saw `keychain_name` from doc but not sure if there is a way to get around that?

**Context -** 

We are using Fastlane to install certs on our CI.

However Fastlane keeps using the `"/Library/Keychains/System.keychain"` when installing certs to, where the certs need to be installed to `"/Users/buildkite-agent/Library/Keychains/login-db"`.

I tried executing fastlane command as the user I wish Fastlane would install the keys to
```
sudo -i -u buildkite-agent bash << EOF
fastlane --verbose get_all_certificates
EOF
```

but it somehow doesn't work, interestingly if I ssh into the machine and
```
sudo su - buildkite-agent
fastlane --verbose get_all_certificates
```

then it works. So I am wondering maybe Fastlane behaves differently under non-login/interactive shell?

**Comment by iBotPeaches** [comment]
Confused on your question. `match` default is `login.keychain` in my memory and was configurable via
 * `keychain_name`
 * `keychain_password`
 
 Already as existing parameters. 
 
 https://docs.fastlane.tools/actions/match/

**Reply by OliverKoo** [reply]
> Confused on your question. `match` default is `login.keychain` in my memory and was configurable via
> 
> * `keychain_name`
> * `keychain_password`
> 
> Already as existing parameters.
> 
> https://docs.fastlane.tools/actions/match/

right but it would deafult to `$HOME/Library/Keychains/login.keychain`. and I assume if I am user "buildkite-agent" the key would be install to `/Users/buildkite-agent/Library/Keychains/login.keychain` but the keys and certs are instead being install to `/Library/Keychains/System.keychain`.....

hence why I am wondering can I specify the full path not just `keychain_name` and `keychain_password`

**Comment by beinginorbit** [comment]
It's bonkers that this conversation just ended here what did you come up with @OliverKoo ? Any lucky?

**Comment by Vespion** [comment]
To anyone still coming across this, keychain_name will be used by match to search the default locations ($HOME/Library/Keychains/<keychain_name>, etc) and if that fails it will then attempt  resolve it as a file using Ruby's default path logic.

I found it most successful when you supply it as an absolute path
